apps/web/components/DatePickerWithRange.tsx

```diff
@@ -13,48 +12,100 @@ import {
   PopoverContent,
   PopoverTrigger,
 } from "@/components/ui/popover";
+import { List } from "@/components/common/List";
+import { differenceInDays, subDays } from "date-fns";
+import { useMemo } from "react";
+
+interface DatePickerWithRangeProps
+  extends React.HTMLAttributes<HTMLDivElement> {
+  dateRange?: DateRange;
+  onSetDateRange: (dateRange?: DateRange) => void;
+  selectOptions: { label: string; value: string }[];
+  dateDropdown: string;
+  onSetDateDropdown: (option: { label: string; value: string }) => void;
+}
 
 export function DatePickerWithRange({
   dateRange,
   onSetDateRange,
-}: React.HTMLAttributes<HTMLDivElement> & {
-  dateRange?: DateRange;
-  onSetDateRange: (dateRange?: DateRange) => void;
-}) {
+  selectOptions,
+  dateDropdown,
+  onSetDateDropdown,
+}: DatePickerWithRangeProps) {
+  const now = useMemo(() => new Date(), []);
+  const days =
+    dateRange?.from && dateRange?.to
+      ? differenceInDays(dateRange.to, dateRange.from)
+      : 0;
+
+  const relativeDateLabel =
+    days === 1
+      ? "Last day"
+      : days === 7
+        ? "Last week"
+        : days === 30
+          ? "Last month"
```

Comment:
This is really ugly. You can probably have a cleaner result with a function. Rough idea:

```tsx
getLabel = () => {
if (days === 1) return "Last day"
if (days === 7) return "Last week"
}
```

Or a switch statement in that func if you prefer.